yt/utilities/lib/pyoctree_raytracing.py

```diff
@@ -0,0 +1,101 @@
+from itertools import product
+
+import numpy as np
+
+from yt.utilities.lib.cyoctree_raytracing import CythonOctreeRayTracing
+
+
+class OctreeRayTracing(object):
+    octree = None
+    data_source = None
+    log_fields = None
+    fields = None
+
+    # Internal data
+    _cell_index = None
+    _tvalues = None
+
+    def __init__(self, data_source):
+        self.data_source = data_source
+        LE = np.array([0, 0, 0], dtype=np.float64)
+        RE = np.array([1, 1, 1], dtype=np.float64)
+        depth = data_source.ds.parameters["levelmax"]
+
+        self.octree = CythonOctreeRayTracing(LE, RE, depth)
```

Comment:
It _is_ very odd. We should discuss better options. @matthewturk @munkm what's your opinion? The issue is that my implementation is multi-layered, there's a C++ implementation that's wrapped in a cython file that's wrapped in a python file... So we could probably get rid of most of these layers, but which ones?

Reply:
I changed the naming so that the python file is named `octree_raytracing.py`, the cython wrapper is `_octree_raytracing.p{xd,yx}` and the C++ file is `_octree_raytracing.hpp`.